app/controllers/cloud_volume_controller.rb

```diff
@@ -419,14 +419,15 @@ def delete_volumes
       volume = CloudVolume.find_by_id(v)
       if volume.nil?
         add_flash(_("%{model} no longer exists.") % {:model => ui_lookup(:table => "cloud_volume")}, :error)
-      elsif !volume.attachments.length.empty?
-        add_flash(_("%{model} \"%{name}\" cannot be removed because it has attachments") % {
-          :model => ui_lookup(:table => 'cloud_volume'),
-          :name  => volume.name}, :warning)
+      elsif !volume.attachments.empty?
+        add_flash(_("%{model} \"%{name}\" cannot be removed because it is attached to one or more %{instances}") % {
+          :model     => ui_lookup(:table => 'cloud_volume'),
+          :name      => volume.name,
+          :instances => ui_lookup(:tables => 'vm_cloud')}, :warning)
       else
         valid_delete, delete_details = volume.validate_delete_volume
```

Comment:
That makes sense, although as you point out below, this is a bug fix to actually make it work, and this validation is here already, but it could be refactored as a future fix.